### PR TITLE
Direct link to vendor list view

### PR DESCRIPTION
**Pull request by skoklowski**

### CODE REVIEW

File: src/components/popup/details/details.jsx [hunk 1]
```diff
@@ -45,30 +52,26 @@ export default class Details extends Component {
 	};
 
 	handleShowVendors = (filter) => {
-		this.state.vendors = this.getVendors(filter);
-		this.setState({
-			selectedPanelIndex: SECTION_VENDORS
-		});
+		this.state.vendors = this.filterVendors(filter);
```
Comment (mvciekzvbek) [inline src/components/popup/details/details.jsx]: Jest jakiś powód, że ustawiasz stan w ten sposób? 
https://reactjs.org/docs/state-and-lifecycle.html#using-state-correctly

Comment (skoklowski) [reply src/components/popup/details/details.jsx]: No, there is no reason. Simply it is remaining from refactoring.

File: src/components/popup/popup.jsx [hunk 2]
```diff
@@ -32,27 +18,24 @@ export default class Popup extends Component {
 	};
 
 	onCancel = () => {
-		this.setState({
-			selectedPanelIndex: SECTION_INTRO
-		});
+		this.props.store.section = SECTION_INTRO;
+		this.forceUpdate();
```
Comment (osk-8) [inline src/components/popup/popup.jsx]: Maybe we should simply use `this.props.store.updateSection(SECTION_INTRO)` in order to omit `forceUpdate()` usage.

File: src/components/popup/popup.jsx [hunk 3]
```diff
@@ -32,27 +18,24 @@ export default class Popup extends Component {
...
 	};
 
 	handleShowDetails = () => {
-		this.setState({
-			selectedPanelIndex: SECTION_DETAILS
-		});
+		this.props.store.section = SECTION_DETAILS;
+		this.forceUpdate();
```
Comment (osk-8) [inline src/components/popup/popup.jsx]: Same here: `this.props.store.updateSection(SECTION_DETAILS)`.

File: src/components/popup/popup.jsx [hunk 4]
```diff
@@ -32,27 +18,24 @@ export default class Popup extends Component {
 	};
 
 	onCancel = () => {
-		this.setState({
```
Comment (mvciekzvbek) [inline src/components/popup/popup.jsx:35]: I would also reconsider modifying props at the expense of removing component's state. 
https://reactjs.org/docs/components-and-props.html#props-are-read-only

File: src/components/popup/details/details.jsx [hunk 5]
```diff
@@ -45,30 +52,27 @@ export default class Details extends Component {
...
-		const { onCancel } = this.props;
-		const { selectedPanelIndex } = this.state;
-		this.setState({
-			selectedPanelIndex: Math.max(0, selectedPanelIndex - 1)
-		});
-		if (selectedPanelIndex === SECTION_PURPOSES) {
+		const { onCancel, store } = this.props;
+		const { subsection } = store;
+		store.subsection = Math.max(0, subsection - 1);
+		this.forceUpdate();
```
Comment (osk-8) [inline src/components/popup/details/details.jsx]: Here we can use `store.updateSection(SECTION_DETAILS, Math.max(0, subsection - 1));`

File: src/components/popup/popup.jsx [hunk 6]
```diff
@@ -32,27 +18,24 @@ export default class Popup extends Component {
...
-			selectedPanelIndex: SECTION_INTRO
-		});
+		this.props.store.section = SECTION_INTRO;
+		this.forceUpdate();
 	};
 
 	handleShowDetails = () => {
-		this.setState({
```
Comment (mvciekzvbek) [inline src/components/popup/popup.jsx:41]: Same case here.

File: src/components/popup/details/details.jsx [hunk 7]
```diff
@@ -45,30 +52,27 @@ export default class Details extends Component {
...
 	handleBack = () => {
-		const { onCancel } = this.props;
-		const { selectedPanelIndex } = this.state;
-		this.setState({
-			selectedPanelIndex: Math.max(0, selectedPanelIndex - 1)
-		});
-		if (selectedPanelIndex === SECTION_PURPOSES) {
+		const { onCancel, store } = this.props;
+		const { subsection } = store;
+		store.subsection = Math.max(0, subsection - 1);
```
Comment (mvciekzvbek) [inline src/components/popup/details/details.jsx]: And here.

File: src/components/popup/details/details.jsx [hunk 8]
```diff
@@ -45,30 +52,27 @@ export default class Details extends Component {
 	};
 
 	handleShowVendors = (filter) => {
-		this.state.vendors = this.getVendors(filter);
+		this.props.store.subsection = SECTION_VENDORS;
```
Comment (osk-8) [inline src/components/popup/details/details.jsx]: Here we can use `this.props.store.updateSection(SECTION_DETAILS, SECTION_VENDORS);`

**Review by osk-8 — APPROVED** [review]

File: src/components/popup/popup.jsx (hunk 4 above)
Comment (skoklowski) [reply src/components/popup/popup.jsx:35]: Ya are right. I have rewritten code and not it uses `store.updateSection` function so `props` are not modified directly by component.

**Review by mvciekzvbek — APPROVED** [review]

**Pull request merged**